ClangSharpPInvokeGenerator/OutputBuilder.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace ClangSharpPInvokeGenerator
+{
+    internal sealed class OutputBuilder : IDisposable
```

Comment:
This could also allow some other interesting configuration options. Such as sorting types or methods before writing them out to the file (which would allow determinism for single-file generation and for the method class file).